vm/src/stdlib/os.rs

```diff
@@ -356,12 +356,30 @@ mod _os {
         }
     }
 
+    #[cfg(any(unix, windows, target_os = "wasi"))]
+    #[derive(FromArgs)]
+    struct OpenArgs {
+        #[pyarg(any)]
+        path: PyPathLike,
+        #[pyarg(any)]
+        flags: OpenFlags,
+        #[pyarg(any, default)]
+        mode: Option<PyIntRef>,
+        #[pyarg(flatten)]
+        dir_fd: DirFd,
+    }
+
     #[cfg(any(unix, windows, target_os = "wasi"))]
     #[pyfunction]
-    pub(crate) fn open(
+    fn open(args: OpenArgs, vm: &VirtualMachine) -> PyResult<i64> {
+        rust_open(args.path, args.flags, args.mode, args.dir_fd, vm)
+    }
+
+    #[cfg(any(unix, windows, target_os = "wasi"))]
+    pub(crate) fn rust_open(
```

Comment:
Maybe you could rename this to `os_open`, but export it below as `pub(crate) use _os::os_open as open;`? This sorta follows the convention of `io_open` in `io.rs`.

Reply:
I am actually wondering about the name `rust_open`, and I think your choice is better. It is now changed.